### ISSUE TITLE
sui select option how to show text e.g <option value="one"> NameOne</option>

### ISSUE THREAD

**Issue by irfanmayo0786** [edcarroll/ng2-semantic-ui#291]
`<option value="one"> NameOne</option>`

how to put NameOne in options?...

**Comment by irfanmayo0786** [comment]
can any one tell me how to show NameOne instead of value in drop down selection?..

**Comment by trevordaniels** [comment]
check out the docs page which shows how to use labelField or templates to achieve this

https://edcarroll.github.io/ng2-semantic-ui/#/modules/select

**Comment by irfanmayo0786** [comment]
Nothing like that i saw the doc can you tell me which one it's is?

**Comment by trevordaniels** [comment]
http://plnkr.co/edit/I3JQpmAdvrsZP5OX5N5K

**Comment by irfanmayo0786** [comment]
we can not give it directly mean static html? im my Case i have like that
`<sui-select formControlName="questionType" (ngModelChange)="onChange($event,i,j)">  
                        <sui-select-option value="radio" labelField="Choose">
                        </sui-select-option>
                          <sui-select-option value="checkbox" labelField="Choose"></sui-select-option>
                          <sui-select-option value="input"></sui-select-option>
                          <sui-select-option value="date">Date</sui-select-option>
                      </sui-select>`

**Comment by trevordaniels** [comment]
No, you cannot insert html directly here. You could use templates or a formatter function as outlined in the docs, or check out this pr to see what's coming soon https://github.com/edcarroll/ng2-semantic-ui/pull/225

**Comment by irfanmayo0786** [comment]
hello i'm using formControlName reactive form instead of the ngModel how can i bind it? still in my case when i bind it with formControlName it bind complete object of option like {name:'radio',value:'radio'} how can i solve it?

**Comment by irfanmayo0786** [comment]
can you tell me how i can use formControlName instead of the [(ngModel)]

**Comment by irfanmayo0786** [comment]
@trevordaniels  i'm waiting your response on it.

**Comment by trevordaniels** [comment]
Reactive forms shouldn't make any difference. Create a plunkr with your
code and I'll see if I can help you.

On 8 Sep 2017 8:49 PM, "irfanmayo0786" <notifications@github.com> wrote:

@trevordaniels <https://github.com/trevordaniels> i'm waiting your response
on it.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/edcarroll/ng2-semantic-ui/issues/291#issuecomment-328072062>,
or mute the thread

**Comment by irfanmayo0786** [comment]
`                  <sui-select class="selection"
                    formControlName="questionType" 
                    [options]="options"
                    labelField="name"
                    [isSearchable]="searchable"
                    [isDisabled]="disabled"
                    #select>
                      <sui-select-option *ngFor="let option of select.filteredOptions"
                                         [value]="option">
                      </sui-select-option>
                  </sui-select>
`
this is my code i'm viewing that question type return {name:'ks',vlaue:'ksl'}

**Comment by trevordaniels** [comment]
Check out the plunkr I sent for how to use valueField

**Comment by irfanmayo0786** [comment]
where is the plunkr url i cannot see it bro. @trevordaniels

**Comment by irfanmayo0786** [comment]
bro i checked it was for not a reactive forms. in reactive forms i got both values. key & name mean i got object in reactive forms.

**Comment by trevordaniels** [comment]
It makes no difference if you're using reactive forms. Use valueField to
return the property you want instead of the whole object. Eg.
valueField="prop"

**Comment by dev-fullstacktechnology** [comment]
Use like this if you have object and you want to return string value,

`<sui-select class="selection form-control"
                          [options]="selectMarketplaceOption"
                          [(ngModel)]="listMarketPlaceDetail.marketplaceId"
                          labelField="market"
                          valueField="id"
                          #marketSelect>
                <sui-select-option *ngFor="let option of marketSelect.filteredOptions"
                                   [value]="option">
                </sui-select-option>
              </sui-select>`

and my Array<Object> = [
    {market: 'abc', id: 'd4rf42'},
    {market: 'def', id: '7yh9y5'}
]

And It returns only id value.
